#Run this script on a high performance cluster.

#loading libraries
library(tidyverse)

#loading data

# To generate the paf files, we aligned cDNA nanopore reads to a merge of the ensemble transcriptome (Homo_sapiens.GRCh38.cdna.all, release 104), 
# ensemble ncRNA (Homo_sapiens.GRCh38.ncrna, release 104), ribosomal RNA and the EF1a-dEGFP-IRIS-PuroR construct using minimap2, 
# with the -ax map-ont arguments. Output .sam files were converted to .paf files using paftools.

##load the paf files generated by minimap2. Adjust file path and name to your situation. 
paf_bc1 <- read_tsv("/your/file/location/BC01_tt.paf", 
                    col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                  "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                  "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc2 <- read_tsv("/your/file/location/BC02_tt.paf", 
                    col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                  "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                  "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc3 <- read_tsv("/your/file/location/BC03_tt.paf", 
                    col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                  "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                  "cigar_string")) %>%
  dplyr::rename("enst" = "target_name")

paf_bc4 <- read_tsv("/your/file/location/BC04_tt.paf", 
                    col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                  "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                  "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc5 <- read_tsv("/your/file/location/BC05_tt.paf", 
                    col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                  "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                  "cigar_string")) %>%
  dplyr::rename("enst" = "target_name")

paf_bc6 <- read_tsv("/your/file/location/BC06_tt.paf", 
                    col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                  "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                  "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc7 <- read_tsv("/your/file/location/BC07_tt.paf", 
                    col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                  "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                  "cigar_string")) %>%
  dplyr::rename("enst" = "target_name")

paf_bc8 <- read_tsv("/your/file/location/BC08_tt.paf", 
                    col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                  "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                  "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc9 <- read_tsv("/your/file/location/BC09_tt.paf", 
                    col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                  "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                  "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc10 <- read_tsv("/your/file/location/BC10_tt.paf", 
                     col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                   "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                   "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc11 <- read_tsv("/your/file/location/BC11_tt.paf", 
                     col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                   "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                   "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc12 <- read_tsv("/your/file/location/BC12_tt.paf", 
                     col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                   "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                   "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc13 <- read_tsv("/your/file/location/BC13_tt.paf", 
                     col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                   "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                   "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc14 <- read_tsv("/your/file/location/BC14_tt.paf", 
                     col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                   "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                   "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc15 <- read_tsv("/your/file/location/BC15_tt.paf", 
                     col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                   "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                   "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

paf_bc16 <- read_tsv("/your/file/location/BC16_tt.paf", 
                     col_names = c("query_name", "query_length", "query_start", "query_end", "strand", "target_name", "target_length", 
                                   "target_start", "target_end", "num_match", "num_map", "mapq", "type_of_aln", "mm_i", "gn_i", "go_i", 
                                   "cigar_string")) %>%
  dplyr::rename("enst" = "target_name") 

### Combine only for selection purposes. Only calculate coverage for ENSTs with an average of over 10 reads per sample. 
### Because low number of reads makes is useless to calculate coverage anyway.
all_paf <- bind_rows(paf_bc1, paf_bc2, paf_bc3, paf_bc4, paf_bc5, paf_bc6, paf_bc7, paf_bc8, 
                    paf_bc9, paf_bc10, paf_bc11, paf_bc12, paf_bc13, paf_bc14, paf_bc15, paf_bc16)

select_enst <- all_paf %>%
  group_by(enst) %>%
  summarise(count = n()) %>%
  filter(count > 160)

rm(all_paf)

### filter out rare transcripts with the selection from above

paf_bc1 <- paf_bc1 %>%
  inner_join(select_enst) %>%
  split(.$enst) # split the table into a list of separate tables per enst

paf_bc2 <- paf_bc2 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc3 <- paf_bc3 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc4 <- paf_bc4 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc5 <- paf_bc5 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc6 <- paf_bc6 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc7 <- paf_bc7 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc8 <- paf_bc8 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc9 <- paf_bc9 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc10 <- paf_bc10 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc11 <- paf_bc11 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc12 <- paf_bc12 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc13 <- paf_bc13 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc14 <- paf_bc14 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc15 <- paf_bc15 %>%
  inner_join(select_enst)%>%
  split(.$enst)

paf_bc16 <- paf_bc16 %>%
  inner_join(select_enst)%>%
  split(.$enst)


# calculating coverage

## make empty results lists

cov_list_bc1 <- vector("list", length = length(paf_bc1))
names(cov_list_bc1) <- names(paf_bc1)

cov_list_bc2 <- vector("list", length = length(paf_bc2))
names(cov_list_bc2) <- names(paf_bc2)

cov_list_bc3 <- vector("list", length = length(paf_bc3))
names(cov_list_bc3) <- names(paf_bc3)

cov_list_bc4 <- vector("list", length = length(paf_bc4))
names(cov_list_bc4) <- names(paf_bc4)

cov_list_bc5 <- vector("list", length = length(paf_bc5))
names(cov_list_bc5) <- names(paf_bc5)

cov_list_bc6 <- vector("list", length = length(paf_bc6))
names(cov_list_bc6) <- names(paf_bc6)

cov_list_bc7 <- vector("list", length = length(paf_bc7))
names(cov_list_bc7) <- names(paf_bc7)

cov_list_bc8 <- vector("list", length = length(paf_bc8))
names(cov_list_bc8) <- names(paf_bc8)

cov_list_bc9 <- vector("list", length = length(paf_bc9))
names(cov_list_bc9) <- names(paf_bc9)

cov_list_bc10 <- vector("list", length = length(paf_bc10))
names(cov_list_bc10) <- names(paf_bc10)

cov_list_bc11 <- vector("list", length = length(paf_bc11))
names(cov_list_bc11) <- names(paf_bc11)

cov_list_bc12 <- vector("list", length = length(paf_bc12))
names(cov_list_bc12) <- names(paf_bc12)

cov_list_bc13 <- vector("list", length = length(paf_bc13))
names(cov_list_bc13) <- names(paf_bc13)

cov_list_bc14 <- vector("list", length = length(paf_bc14))
names(cov_list_bc14) <- names(paf_bc14)

cov_list_bc15 <- vector("list", length = length(paf_bc15))
names(cov_list_bc15) <- names(paf_bc15)

cov_list_bc16 <- vector("list", length = length(paf_bc16))
names(cov_list_bc16) <- names(paf_bc16)

### fill results lists with empty table
for (name in names(paf_bc1)){
  enst_length <- paf_bc1[[name]]$target_length[[1]]
  cov_list_bc1[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc2)){
  enst_length <- paf_bc2[[name]]$target_length[[1]]
  cov_list_bc2[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc3)){
  enst_length <- paf_bc3[[name]]$target_length[[1]]
  cov_list_bc3[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc4)){
  enst_length <- paf_bc4[[name]]$target_length[[1]]
  cov_list_bc4[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc5)){
  enst_length <- paf_bc5[[name]]$target_length[[1]]
  cov_list_bc5[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc6)){
  enst_length <- paf_bc6[[name]]$target_length[[1]]
  cov_list_bc6[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc7)){
  enst_length <- paf_bc7[[name]]$target_length[[1]]
  cov_list_bc7[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc8)){
  enst_length <- paf_bc8[[name]]$target_length[[1]]
  cov_list_bc8[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc9)){
  enst_length <- paf_bc9[[name]]$target_length[[1]]
  cov_list_bc9[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc10)){
  enst_length <- paf_bc10[[name]]$target_length[[1]]
  cov_list_bc10[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc11)){
  enst_length <- paf_bc11[[name]]$target_length[[1]]
  cov_list_bc11[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc12)){
  enst_length <- paf_bc12[[name]]$target_length[[1]]
  cov_list_bc12[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc13)){
  enst_length <- paf_bc13[[name]]$target_length[[1]]
  cov_list_bc13[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc14)){
  enst_length <- paf_bc14[[name]]$target_length[[1]]
  cov_list_bc14[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc15)){
  enst_length <- paf_bc15[[name]]$target_length[[1]]
  cov_list_bc15[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}
for (name in names(paf_bc16)){
  enst_length <- paf_bc16[[name]]$target_length[[1]]
  cov_list_bc16[[name]] <- tibble(position = seq_len(enst_length), cov_count = 0, n_reads = 0)
}

## calculate coverages 1
for (name in names(paf_bc1)){
  for (row in seq_len(nrow(paf_bc1[[name]]))) {
    cov_list_bc1[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc1[[name]][row,]$target_start <= position & paf_bc1[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc1[[name]]
  }
}

### unlist results
cov_list_bc1 <- cov_list_bc1 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 0,
         treatment = "no_transduction")

write_tsv(cov_list_bc1, file = "cov_list_bc1.tsv")

## calculate coverages 2
for (name in names(paf_bc2)){
  for (row in seq_len(nrow(paf_bc2[[name]]))) {
    cov_list_bc2[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc2[[name]][row,]$target_start <= position & paf_bc2[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc2[[name]]
  }
}

### unlist results
cov_list_bc2 <- cov_list_bc2 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 0,
         treatment = "empty_transduction")

write_tsv(cov_list_bc2, file = "cov_list_bc2.tsv")

## calculate coverages 3
for (name in names(paf_bc3)){
  for (row in seq_len(nrow(paf_bc3[[name]]))) {
    cov_list_bc3[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc3[[name]][row,]$target_start <= position & paf_bc3[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc3[[name]]
  }
}

### unlist results
cov_list_bc3 <- cov_list_bc3 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 0,
         treatment = "non_targeting")

write_tsv(cov_list_bc3, file = "cov_list_bc3.tsv")

## calculate coverages 4
for (name in names(paf_bc4)){
  for (row in seq_len(nrow(paf_bc4[[name]]))) {
    cov_list_bc4[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc4[[name]][row,]$target_start <= position & paf_bc4[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc4[[name]]
  }
}

### unlist results
cov_list_bc4 <- cov_list_bc4 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 0,
         treatment = "targeting_sp2")

write_tsv(cov_list_bc4, file = "cov_list_bc4.tsv")

## calculate coverages 5
for (name in names(paf_bc5)){
  for (row in seq_len(nrow(paf_bc5[[name]]))) {
    cov_list_bc5[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc5[[name]][row,]$target_start <= position & paf_bc5[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc5[[name]]
  }
}

### unlist results
cov_list_bc5 <- cov_list_bc5 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 50,
         treatment = "empty_transduction")

write_tsv(cov_list_bc5, file = "cov_list_bc5.tsv")

## calculate coverages 6
for (name in names(paf_bc6)){
  for (row in seq_len(nrow(paf_bc6[[name]]))) {
    cov_list_bc6[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc6[[name]][row,]$target_start <= position & paf_bc6[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc6[[name]]
  }
}

### unlist results
cov_list_bc6 <- cov_list_bc6 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 50,
         treatment = "non_targeting")

write_tsv(cov_list_bc6, file = "cov_list_bc6.tsv")

## calculate coverages 7
for (name in names(paf_bc7)){
  for (row in seq_len(nrow(paf_bc7[[name]]))) {
    cov_list_bc7[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc7[[name]][row,]$target_start <= position & paf_bc7[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc7[[name]]
  }
}

### unlist results
cov_list_bc7 <- cov_list_bc7 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 50,
         treatment = "targeting_sp2")

write_tsv(cov_list_bc7, file = "cov_list_bc7.tsv")

## calculate coverages
for (name in names(paf_bc8)){
  for (row in seq_len(nrow(paf_bc8[[name]]))) {
    cov_list_bc8[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc8[[name]][row,]$target_start <= position & paf_bc8[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc8[[name]]
  }
}

### unlist results
cov_list_bc8 <- cov_list_bc8 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 200,
         treatment = "empty_transduction")

write_tsv(cov_list_bc8, file = "cov_list_bc8.tsv")

## calculate coverages 9
for (name in names(paf_bc9)){
  for (row in seq_len(nrow(paf_bc9[[name]]))) {
    cov_list_bc9[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc9[[name]][row,]$target_start <= position & paf_bc9[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc9[[name]]
  }
}

### unlist results
cov_list_bc9 <- cov_list_bc9 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 200,
         treatment = "non_targeting")

write_tsv(cov_list_bc9, file = "cov_list_bc9.tsv")

## calculate coverages 10
for (name in names(paf_bc10)){
  for (row in seq_len(nrow(paf_bc10[[name]]))) {
    cov_list_bc10[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc10[[name]][row,]$target_start <= position & paf_bc10[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc10[[name]]
  }
}

### unlist results
cov_list_bc10 <- cov_list_bc10 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 200,
         treatment = "targeting_sp2")

write_tsv(cov_list_bc10, file = "cov_list_bc10.tsv")

## calculate coverages 11
for (name in names(paf_bc11)){
  for (row in seq_len(nrow(paf_bc11[[name]]))) {
    cov_list_bc11[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc11[[name]][row,]$target_start <= position & paf_bc11[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc11[[name]]
  }
}

### unlist results
cov_list_bc11 <- cov_list_bc11 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 500,
         treatment = "empty_transduction")

write_tsv(cov_list_bc11, file = "cov_list_bc11.tsv")

## calculate coverages 12
for (name in names(paf_bc12)){
  for (row in seq_len(nrow(paf_bc12[[name]]))) {
    cov_list_bc12[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc12[[name]][row,]$target_start <= position & paf_bc12[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc12[[name]]
  }
}

### unlist results
cov_list_bc12 <- cov_list_bc12 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 500,
         treatment = "non_targeting")

write_tsv(cov_list_bc12, file = "cov_list_bc12.tsv")

## calculate coverages 13
for (name in names(paf_bc13)){
  for (row in seq_len(nrow(paf_bc13[[name]]))) {
    cov_list_bc13[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc13[[name]][row,]$target_start <= position & paf_bc13[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc13[[name]]
  }
}

### unlist results
cov_list_bc13 <- cov_list_bc13 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 500,
         treatment = "targeting_sp2")

write_tsv(cov_list_bc13, file = "cov_list_bc13.tsv")

## calculate coverages 14
for (name in names(paf_bc14)){
  for (row in seq_len(nrow(paf_bc14[[name]]))) {
    cov_list_bc14[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc14[[name]][row,]$target_start <= position & paf_bc14[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc14[[name]]
  }
}

### unlist results
cov_list_bc14 <- cov_list_bc14 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 1440,
         treatment = "empty_transduction")

write_tsv(cov_list_bc14, file = "cov_list_bc14.tsv")

## calculate coverages 15
for (name in names(paf_bc15)){
  for (row in seq_len(nrow(paf_bc15[[name]]))) {
    cov_list_bc15[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc15[[name]][row,]$target_start <= position & paf_bc15[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc15[[name]]
  }
}

### unlist results
cov_list_bc15 <- cov_list_bc15 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 1440,
         treatment = "non_targeting")

write_tsv(cov_list_bc15, file = "cov_list_bc15.tsv")

## calculate coverages 16
for (name in names(paf_bc16)){
  for (row in seq_len(nrow(paf_bc16[[name]]))) {
    cov_list_bc16[[name]] %>%
      mutate(n_reads = n_reads + 1,
             cov_count = if_else(paf_bc16[[name]][row,]$target_start <= position & paf_bc16[[name]][row,]$target_end >= position,
                                 cov_count + 1, cov_count + 0)) -> cov_list_bc16[[name]]
  }
}

### unlist results
cov_list_bc16 <- cov_list_bc16 %>%
  bind_rows(.id = "enst") %>%
  mutate(timepoint = 1440,
         treatment = "targeting_sp2")

write_tsv(cov_list_bc16, file = "cov_list_bc16.tsv")

#merging and saving final table

all_cov <- bind_rows(cov_list_bc1, cov_list_bc2, cov_list_bc3, cov_list_bc4,
                     cov_list_bc5, cov_list_bc6, cov_list_bc7, cov_list_bc8,
                     cov_list_bc9, cov_list_bc10, cov_list_bc11, cov_list_bc12,
                     cov_list_bc13, cov_list_bc14, cov_list_bc15, cov_list_bc16)

write_tsv(all_cov, file = "all_cov.tsv")

